### ISSUE TITLE
Discrepancy between training results and provided checkpoint

### ISSUE THREAD

**Issue by halsstarrig243** [zju3dv/StreamingTalker#3]
Hello Yifan Yang.
First of all, thank you for your impressive work on StreamingTalker and for sharing the code with the community.
Context & Problem Description: I am currently trying to reproduce the results using the training configuration files provided in the repository. Specifically, we utilized the pre-trained VQ-VAE weights provided by you to train the generator.
However, after attempting to train the model multiple times following the instructions, I found that the results were not ideal and showed a significant discrepancy compared to the inference results obtained using your provided checkpoint.
Observations: I have attached the metric comparisons below for reference.
Figure 1 (Our Training): Results obtained by training the generator from scratch (using your pre-trained VQ-VAE) with the provided config.
Note: As seen in the screenshot, the MOD (Motion metrics) is lower (0.0037 vs 0.0054), potentially indicating less expressive facial movements compared to the official weights.
Figure 2 (Official Checkpoint): Results obtained by running inference on your released checkpoint.
<img width="705" height="284" alt="Image" src="https://github.com/user-attachments/assets/9c1a1657-7348-4211-a12d-48d99dcdf59e" />
<img width="677" height="290" alt="Image" src="https://github.com/user-attachments/assets/a94b7e3e-c4e8-49ae-93a8-bef54e4d3aeb" />
Questions: Could you please clarify the following:
Are the configuration files currently in the repository (e.g., config.yaml or hyperparameters) exactly the same as the ones used to train the released checkpoint?
If there are any differences, could you kindly provide the specific parameter file or share details regarding the hyperparameters (such as learning rate schedule, batch size, or total steps) used for the best checkpoint?
Any guidance on how to close this gap would be greatly appreciated.
Thank you very much for your time and help!

**Comment by yangyifan18** [comment]
Hello, thanks for your attention! 
Before the code release, I re-trained the model for multiple times based on configuration files currently in the repository, and get the results below (Most of the time, the result is close to the best metrics showed in paper)
<img width="1151" height="312" alt="Image" src="https://github.com/user-attachments/assets/e4d73986-3be7-45c6-935a-6d9caa90f701" />

I will repeat more times to check if it's our model's instability to randomness. This may take 2-3 days.
Meanwhile, I'll compare the configuration files with ones I previously used.

And this is test result on our provided ckpt:
<img width="513" height="142" alt="Image" src="https://github.com/user-attachments/assets/996f3efc-e7f7-4477-a836-14f913a1ca4e" />

Please check:
Did you use the preprocessed dataset downloaded from modelscope? Or you processed the dataset by yourself?
Could you paste the yaml you used for training?

Best wishes.

**Comment by halsstarrig243** [comment]
Hello @yangyifan18,

Thank you so much for your patience and for taking the time to re-verify the training results.

Following your suggestion, we re-downloaded the preprocessed dataset (from ModelScope) and set up the environment again strictly according to the documentation. After re-training, we found that our results are now very consistent with the re-training metrics you just shared.

As shown in the screenshot below, our new run achieved an MOD of approximately 0.00389, which aligns closely with your result (~0.00380).

<img width="921" height="537" alt="Image" src="https://github.com/user-attachments/assets/bdbb9aad-9f3a-43e3-aa93-77ca07ce2760" />

Thank you again for your guidance and for this excellent work!

Best wishes.

**Comment by yangyifan18** [comment]
You are welcome!
In practice, you can also try more epochs (for example, 600) to get better results. I'll update the config yaml files in repo later.
